### PR TITLE
Feature/reference alleles

### PR DESCRIPTION
**Pull request by luissian**
Implementation of this feature is done by creating 2 additional classes: for getting the candidate alleles 
- using distance matrix(mash)
- clustering (using  Leigde algorithm) 

On this PR , reference alleles are selected by choosing an allele for each cluster. The cluster is done using Leigde algorithm that get as input the distance matrix created by mash program. Resolution parameter is used for Leigde algorithm to group neighborhood alleles. The resolution default or user defined value is increased by 0.025 if exists any cluster that the list of  the cluster are not the same as got by blast.
Generated files are:
- reference alleles
- summary cluster
- evaluation
- statistics graphic to represent the number of locus and number of clusters. for example if on x axis =3 and y axis y = 8  means that there are 3 locus that all alleles are grouped in 8 clusters.
num_genes_per_allele

### CODE REVIEW

File: taranis/distance.py [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+            pd.DataFrame: Triangular distance matrix as panda DataFrame
+        """
+        allele_name = Path(self.file_path).stem
+        mash_distance_command = [
+            "mash",
+            "triangle",
+            "-i",
+            self.file_path,
+            "-k",
+            "17",
```
Comment (saramonzon) [inline taranis/distance.py]: I'll try to include the mash parameters as parameters in case we need to adjust them

File: taranis/clustering.py [hunk 2]
```diff
@@ -0,0 +1,158 @@
...
+            n_iterations=-1,
+            resolution_parameter=self.res_param,
+            seed=self.seed,
+        )
+        cluster_ptrs = np.array(graph_clusters.membership)
+
+        clusters_data = self.collect_data_cluster(cluster_ptrs)
+        # check that cluste average values are upper than 0.9
+        for value in clusters_data.values():
+            if value["avg"] < 0.9:
```
Comment (saramonzon) [inline taranis/clustering.py]: What is this check for? Which is the cluster average value?

File: taranis/clustering.py [hunk 3]
```diff
@@ -0,0 +1,158 @@
...
+
+        Args:
+            dist_matrix (np.array): distance matrix
+            ref_seq_name (str): locus name
+        """
+        self.dist_matrix = dist_matrix
+        self.num_seq = dist_matrix.shape[0]
+        self.ref_seq_name = ref_seq_name
+        self.seed = None
+        self.res_param = 0.9
```
Comment (saramonzon) [inline taranis/clustering.py]: Include this as class param please

File: .github/workflows/tests.yml [hunk 4]
```diff
@@ -26,6 +26,6 @@ jobs:
         run: |
           source $CONDA/etc/profile.d/conda.sh
           conda activate taranis_env
-          poetry install
+          python -m pip install .
           taranis analyze-schema -i  test/MLST_listeria -o analyze_schema_test  --cpus 1 --output-allele-annot --remove-no-cds --remove-duplicated --remove-subset
```
Comment (saramonzon) [inline .github/workflows/tests.yml:30]: please add the test for the reference_alleles functionality

File: setup.py [hunk 5]
```diff
@@ -0,0 +1,37 @@
+#!/usr/bin/env python
+
+from setuptools import setup, find_packages
```
Comment (saramonzon) [inline setup.py:3]: is this file necessary? I thought it was replace by pyproject.toml

File: taranis/clustering.py (hunk 2 above)
Comment (luissian) [reply taranis/clustering.py]: This was for testing propose, to see if any of the cluster had and average bellow 0.9. However during design/testing I  never get an average under 0.9, So I will remove this piece of code, as it is not needed.

File: setup.py (hunk 5 above)
Comment (luissian) [reply setup.py:3]: I put in your mind the idea that setup.py was deprecated.
However I found this link
https://packaging.python.org/en/latest/discussions/setup-py-deprecated/

**Should setup.py be deleted?**

Although the usage of setup.py as an executable script is deprecated, its usage as a configuration file for setuptools is absolutely fine. There is likely no modification needed in setup.py.

**Is pyproject.toml mandatory?**
While it is not technically necessary yet, it is STRONGLY RECOMMENDED for a project to have a pyproject.toml file at the root of its source tree.

When creating the packet now configuration of setup.py is not read, I want to keep it for a while to confirm that we are not using at any time.

File: taranis/distance.py (hunk 1 above)
Comment (luissian) [reply taranis/distance.py]: Yes, I will set as input parameters

File: taranis/clustering.py (hunk 3 above)
Comment (luissian) [reply taranis/clustering.py]: Yes, I will included

File: .github/workflows/tests.yml (hunk 4 above)
Comment (luissian) [reply .github/workflows/tests.yml:30]: I want  to finish coding first, as I  would like to start using pytest for testing

File: setup.py (hunk 5 above)
Comment (saramonzon) [reply setup.py:3]: ok!

**Review by saramonzon — APPROVED** [review]
